packages/web/src/lib/web-mode.ts

```diff
@@ -129,18 +131,22 @@ export class WebMode {
 
 	/**
 	 * Execute single transaction or group of transactions (atomic transaction)
-	 * @param execParams transaction parameters or atomic transaction parameters
+	 * @param transactions transaction parameters, atomic transaction parameters
+	 * or TransactionAndSign object(SDK transaction object and signer parameters)
 	 */
-	async executeTx(execParams: ExecParams[]): Promise<JsonPayload> {
+	async executeTx(
+		transactions: ExecParams[] | TransactionAndSign[]
+	): Promise<algosdk.modelsv2.PendingTransactionResponse> {
 		let txns: Transaction[] = [];
-		if (execParams.length > 16) {
+		if (transactions.length > 16) {
 			throw new Error("Maximum size of an atomic transfer group is 16");
 		}
-
-		for (const [_, txn] of execParams.entries()) {
-			txns.push(mkTransaction(txn, await this.getSuggestedParams(txn.payFlags)));
+		if (!isSDKTransactionAndSign(transactions[0])) {
```

Comment:
add a check before the if:
```
if (transactions.length == 0)
   throw new SomeErrorClass("at least one transaction must be provided")
```

Please find the best error for the error class from the errors we have defined.

Reply:
done